app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt

```diff
@@ -0,0 +1,50 @@
+package org.fossasia.openevent.general.event
+
+import org.fossasia.openevent.general.R
+import org.fossasia.openevent.general.data.Resource
+import org.fossasia.openevent.general.utils.nullToEmpty
+import org.threeten.bp.ZoneId
+import org.threeten.bp.ZonedDateTime
+import org.threeten.bp.format.DateTimeFormatter
+
+object EventUtils {
+
+    @JvmStatic
+    private val sharedResource by lazy {
+        Resource()
+    }
+
+    private val timeFormat: DateTimeFormatter = DateTimeFormatter.ofPattern("hh:mm a")
+    private val dateFormat: DateTimeFormatter = DateTimeFormatter.ofPattern("dd MMM yyyy")
+
+    fun getSharableInfo(event: Event, resource: Resource = sharedResource): String {
+        val description = event.description.nullToEmpty()
+        val eventUrl = event.externalEventUrl.nullToEmpty()
+
+        val message = StringBuilder()
+
+        val startsAt = getLocalizedDateTime(event.startsAt)
+        val endsAt = getLocalizedDateTime(event.endsAt)
+
+        message.append(resource.getString(R.string.event_name)).append(event.name).append("\n\n")
+        if (!description.isEmpty()) message.append(resource.getString(R.string.event_description))
+                .append(event.description.nullToEmpty()).append("\n\n")
+        message.append(resource.getString(R.string.starts_on))
+                .append(startsAt.format(dateFormat)).append(" ")
+                .append(startsAt.format(timeFormat)).append("\n")
+        message.append(resource.getString(R.string.ends_on))
+                .append(endsAt.format(dateFormat)).append(" ")
+                .append(endsAt.format(timeFormat))
```

Comment:
@iamareebjamal So we are assuming that there will be always a start date and an end date?

Reply:
Because the API says so

How can there be an event without start and end date?

Check API documentation

Reply:
@iamareebjamal Yeah of course there can't be :+1: